Azure/azure-cli
# [Compute] `az vm image list`: Add parameter `--architecture` to filter image with its architecture

**Related command**

- `az vm image list`

**Description**

Feature Request #22326

**Testing Guide**

- `azdev test test_vm_image_list_by_alias_and_filtered_by_arch`
- `azdev test test_vm_image_list_thru_services_filtered_by_arch`
- `azdev test test_vm_image_list_thru_services_edge_zone_by_arch`

Due to lack of images with Arm64 architecture in aliases.json, I did some local tests to verify correctness of the filter. Here I changed the local logic to use local aliases, and changed some archtecture to Arm64 in local alias when testing. Below are the test results. 
![image](https://user-images.githubusercontent.com/32201005/175454752-49a6ad28-131a-4bda-8aca-1e841bb71453.png)

**History Notes**


[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.
